apps/aens/src/aens_update_tx.erl

```diff
@@ -0,0 +1,201 @@
+    AccountPubKey.
+
+-spec check(update_tx(), trees(), height()) -> {ok, trees()} | {error, term()}.
+check(#ns_update_tx{account = AccountPubKey, nonce = Nonce,
+                    fee = Fee, name_hash = NameHash, ttl = TTL}, Trees, Height) ->
+    Checks =
+        [fun() -> check_ttl(TTL) end,
+         fun() -> aetx_utils:check_account(AccountPubKey, Trees, Height, Nonce, Fee) end,
+         fun() -> aens_utils:check_name_claimed_and_owned(NameHash, AccountPubKey, Trees) end],
+
+    case aeu_validation:run(Checks) of
+        ok              -> {ok, Trees};
+        {error, Reason} -> {error, Reason}
+    end.
+
+-spec process(update_tx(), trees(), height()) -> {ok, trees()}.
+process(#ns_update_tx{account = AccountPubKey, nonce = Nonce, fee = Fee,
+                      name_hash = NameHash} = UpdateTx, Trees0, Height) ->
+    AccountsTree0 = aec_trees:accounts(Trees0),
+    NSTree0 = aec_trees:ns(Trees0),
+
+    Account0 = aec_accounts_trees:get(AccountPubKey, AccountsTree0),
+    {ok, Account1} = aec_accounts:spend(Account0, Fee, Nonce, Height),
+    AccountsTree1 = aec_accounts_trees:enter(Account1, AccountsTree0),
+
+    Name0 = aens_state_tree:get_name(NameHash, NSTree0),
+    Name1 = aens_names:update(UpdateTx, Name0, Height),
+    NSTree1 = aens_state_tree:enter_name(Name1, NSTree0),
+
+    Trees1 = aec_trees:set_accounts(Trees0, AccountsTree1),
+    Trees2 = aec_trees:set_ns(Trees1, NSTree1),
+
+    {ok, Trees2}.
+
+-spec signers(update_tx()) -> [pubkey()].
+signers(#ns_update_tx{account = AccountPubKey}) ->
+    [AccountPubKey].
+
+-spec serialize(update_tx()) -> list(map()).
+serialize(#ns_update_tx{account   = AccountPubKey,
+                        nonce     = Nonce,
+                        name_hash = NameHash,
+                        name_ttl  = NameTTL,
+                        pointers  = Pointers,
+                        ttl       = TTL,
+                        fee       = Fee}) ->
+    [#{<<"type">>     => type()},
+     #{<<"vsn">>      => version()},
+     #{<<"account">>  => AccountPubKey},
+     #{<<"nonce">>    => Nonce},
+     #{<<"hash">>     => NameHash},
+     #{<<"name_ttl">> => NameTTL},
+     #{<<"pointers">> => jsx:encode(Pointers)},
+     #{<<"ttl">>      => TTL},
+     #{<<"fee">>      => Fee}].
+
+-spec deserialize(list(map())) -> update_tx().
+deserialize([#{<<"type">>     := ?NAME_UPDATE_TX_TYPE},
+             #{<<"vsn">>      := ?NAME_UPDATE_TX_VSN},
+             #{<<"account">>  := AccountPubKey},
+             #{<<"nonce">>    := Nonce},
+             #{<<"hash">>     := NameHash},
+             #{<<"name_ttl">> := NameTTL},
+             #{<<"pointers">> := Pointers},
+             #{<<"ttl">>      := TTL},
+             #{<<"fee">>      := Fee}]) ->
+    #ns_update_tx{account   = AccountPubKey,
+                  nonce     = Nonce,
+                  name_hash = NameHash,
+                  name_ttl  = NameTTL,
+                  pointers  = jsx:decode(Pointers,[{labels, atom}]),
```

Comment:
This looks deserialized from network on external API: could we avoid creating atoms?